File: apps/fogg/src/pump/want/once.go
package want

import (
	"fogg/src/pump/into"
	"fogg/src/type"
	"strings"
)

func Once(options types.Options) bool {
	if options == nil {
		return false
	}
	opt := into.CStr(options)
	return opt == "once" || strings.HasPrefix(opt, "into:") ||
		strings.Contains(opt, `"once":true`) || strings.Contains(opt, `"once": true`)
}

File: apps/boxc/src/pump/want/once.cpp
#include "../../type.hpp"
#include <string_view>

namespace want {
    bool once(Options options) {
        if (!options) return false;
        std::string_view opt(options);
        return opt == "once" || opt.rfind("into:", 0) == 0 ||
               opt.find("\"once\":true") != std::string_view::npos ||
               opt.find("\"once\": true") != std::string_view::npos;
    }
}
